src/ln/channel.rs

```diff
@@ -1883,7 +1889,7 @@ impl Channel {
 	/// waiting on this revoke_and_ack. The generation of this new commitment_signed may also fail,
 	/// generating an appropriate error *after* the channel state has been updated based on the
 	/// revoke_and_ack message.
-	pub fn revoke_and_ack(&mut self, msg: &msgs::RevokeAndACK, fee_estimator: &FeeEstimator) -> Result<(Option<msgs::CommitmentUpdate>, Vec<(PendingForwardHTLCInfo, u64)>, Vec<(HTLCSource, [u8; 32], HTLCFailReason)>, Option<msgs::ClosingSigned>, ChannelMonitor), ChannelError> {
+	pub fn revoke_and_ack(&mut self, msg: &msgs::RevokeAndACK, fee_estimator: &FeeEstimator) -> Result<(Option<msgs::CommitmentUpdate>, Vec<(PendingForwardHTLCInfo, u64)>, Vec<(HTLCSource, [u8; 32], HTLCFailReason)>, Option<msgs::ClosingSigned>, ChannelMonitor, OutPoint), ChannelError> {
```

Comment:
In fact I need to link HTLCSource to a backward ChannelMonitor, so I need funding_txo linked to a given HTLCSource which is itself linked by HTLC hash.
But wouldn't be surprised if there is an easier way ?

Reply:
Maybe need deeper thoughts if we move funding_txo into KeysStorage we maybe don't want also to use it in SimpleManyChannelMonitor to index monitors (but afterwards SimpleManyChannelMonitor is just an implementation of ManyChannelMonitor, not necessarily the watchtower one)

Reply:
The SimpleManyChannelMonitor is intended really just for use for local channels, not for watchtowers, which need to index by a random public key which is used to sign updates from the client (as otherwise your counterparty could replace your monitor with bogus data), so I think we can rely on using the outpoint there. As for linking HTLCSource back to a ChannelMonitor, I think we can just eat the performance hit of storing the short_channel_id along with the funding_txo in the ChannelMonitor. That would let the current HTLCSource structure look up the ChannelMonitor and doing an iteration over all our ChannelMonitors in the case of resolving a on-chain <-> on-chain forward isn't really a big deal, I think.